proposals/dark_mode_extension_icons.md

```diff
@@ -0,0 +1,301 @@
+# Dark Mode Extension Icons
+
+**Summary**
+
+Feature to enable developers to enhance extension icon visibility in dark mode.
+
+**Document Metadata**
+
+**Author:** solomonkinard
+
+**Sponsoring Browser:** Chromium
+
+**Contributors:** oliverdunk, xeenon, carlosjeurissen, hanguokai, dotproto
+
+**Created:** 2024-04-05
+
+**Related Issues:**
+* https://crbug.com/893175
+* https://github.com/w3c/webextensions/issues/229
+
+## Motivation
+
+### Objective
+
+Extension developers will be able to supply and define a set of icons to be used
+in the event that the user has expressed the preference for a page that has a
+dark theme.
+
+#### Use Cases
+
+1. Improved icon visibility in the extension toolbar.
+1. Improved icon visibility on the management and shortcuts pages.
+1. Dark mode icon declarations made possible through the extension manifest.
+1. `setIcon()` will allow setting of dark and/or light mode specific icons.
+1. Improved icon visibility on context menu (Chrome relies on default
+icons, only Firefox and Safari can specify menu icons).
+1. Improved icon visibility on side panel (Chrome relies on default icons,
+Firefox can specify icons).
+
+### Known Consumers
+
+The Chromium bug has a significant amount of developer interest.
+
+## Specification
+
+### Schema
+
+manifest.json
+```
+type ImagePath = string;
+type ColorScheme = "light" | "dark";
+
+interface IconVariantsBase {
+  color_scheme?: ColorScheme | ColorScheme[];
+}
+
+interface IconVariantsManifest extends IconVariantsBase {
+  any?: ImagePath;
```

Comment:
I think this should be something like `[key: string]`? What we have now is a literal key with the name "any".